ui/features/landing/landing.controller.js

```diff
@@ -15,297 +10,331 @@ export default class LandingController {
     this.$rootScope = $rootScope;
     this.model = null;
     this.neighbors = [];
-    var color = {
-        'subClassOf':'#FF4646',
-        '~subClassOf':'#992A2A',
-        'isDefinedBy':'#00FF00',
-        '~isDefinedBy':'#005B00',
-        'equivalentClass':'#5BCBB6',
-        '~equivalentClass':'#2D655B',
-        'sameAs':'#774177',
-        'subPropertyOf':'#FFC04C',
-        '!subPropertyOf':'#A47B30',
-        'equivalentProperty':'#6666FF',
-        '~equivalentProperty':'#3D3D99',
-        'property':'#FF7032',
-        '~property':'#CC3D00',
-        'type':'#B2FFFF',
-        '~type':'#EDD95D',
-        'operand':'#000000'};
-    var duration = 750;
-    var root;
-    var currentNode;
-    var currentColor;
-
-    var d3 = global.d3;
-
-    var margin = {top: 20, right: 120, bottom: 20, left: 20},
-        width = 180,
-        height = 800 - margin.top - margin.bottom;
-
-    var tree = d3.layout.tree()
-        .size([height, width]);
-
-    var diagonal = d3.svg.diagonal()
+    this.d3 = global.d3;
+    this.root = null;
+
+    this.color = {
+        'subClassOf': '#FF4646',
+        '~subClassOf': '#992A2A',
+        'isDefinedBy': '#00FF00',
+        '~isDefinedBy': '#005B00',
+        'equivalentClass': '#5BCBB6',
+        '~equivalentClass': '#2D655B',
+        'sameAs': '#774177',
+        'subPropertyOf': '#FFC04C',
+        '!subPropertyOf': '#A47B30',
+        'equivalentProperty': '#6666FF',
+        '~equivalentProperty': '#3D3D99',
+        'property': '#FF7032',
+        '~property': '#CC3D00',
+        'type': '#B2FFFF',
+        '~type': '#EDD95D',
+        'operand': '#000000'
+    };
+
+    this.duration = 750;
+    this.margin = {top: 20, right: 120, bottom: 20, left: 20};
+    this.width = 180;
+    this.height = 800 - this.margin.top - this.margin.bottom;
+    this.tree = null;
+    this.currentNode = null;
+    this.currentColor = null;
+
+    this.drawBase();
+  }
+
+
+  drawBase() {
+    this.tree = this.d3.layout.tree()
+        .size([this.height, this.width]);
+
+    this.diagonal = this.d3.svg.diagonal()
         .projection(function(d) {
           return [d.y, d.x];
         });
 
-    var svg = d3.select("#ObOne").append("svg")
-        .attr("width", width + margin.right + margin.left)
-        .attr("height", height + margin.top + margin.bottom);
-
-    var graphDraw = svg.append("g")
-        .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+    this.svg = this.d3.select("#ObOne").append("svg")
+        .attr("width", this.width + this.margin.right + this.margin.left)
+        .attr("height", this.height + this.margin.top + this.margin.bottom);
 
+    this.graphDraw = this.svg.append("g")
+        .attr("transform", "translate(" + this.margin.left + "," + this.margin.top + ")");
 
-    var divScroll = document.getElementById("outerOB");
```

Comment:
Totally fine with this change. Just curious why it's good practice to assign it to the instance of the service instead of as a variable.